### PR TITLE
Add safe WebMessageListener calls

### PR DESCRIPTION
**Pull request by joshliebe**
Task/Issue URL: https://app.asana.com/0/1200204095367872/1207622254939926/f

### Description

- Adds safe calls when using `WebMessageListener`.

- Checks:
  - `isFeatureSupported(WebViewFeature.WEB_MESSAGE_LISTENER)`
  - WebView  version >= "126.0.6478.40"
  - `!isDestroyed`
  
- Adds lint checks to enforce usage of these methods.
  
### Steps to test this PR
- [x] Code review

### CODE REVIEW

**Comment by joshliebe** [comment]
* **#4676** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4676?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @joshliebe and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/main/java/com/duckduckgo/app/browser/DuckDuckGoWebView.kt [hunk 1]
```diff
@@ -267,12 +274,55 @@ class DuckDuckGoWebView : WebView, NestedScrollingChild3 {
...
+        webViewVersionProvider: WebViewVersionProvider,
+    ): Boolean {
+        return withContext(dispatchers.io()) {
+            webViewVersionProvider.getFullVersion()
+                .compareSemanticVersion(WEB_MESSAGE_LISTENER_WEBVIEW_VERSION)?.let { it >= 0 } ?: false
+        } && WebViewFeature.isFeatureSupported(WebViewFeature.WEB_MESSAGE_LISTENER)
+    }
+
+    @SuppressLint("RequiresFeature", "AddWebMessageListenerUsage")
+    suspend fun safeAddWebMessageListener(
```
Comment (CDRussell) [inline app/src/main/java/com/duckduckgo/app/browser/DuckDuckGoWebView.kt:288]: We need to offer up some way for the caller to know when this fails. For some features, silently not adding the web message listener is fine, but for others we'll need to take action (show error message in the UI etc..).

Even returning a `Boolean` indicating if it was successfully added would suffice.

Comment (joshliebe) [reply app/src/main/java/com/duckduckgo/app/browser/DuckDuckGoWebView.kt:288]: Good idea! I’ll add that

Comment (joshliebe) [reply app/src/main/java/com/duckduckgo/app/browser/DuckDuckGoWebView.kt:288]: @CDRussell I’ve added the boolean here: https://github.com/duckduckgo/Android/pull/4676/commits/84e71abab78ba762ecc25f207f0e6f0c6d44deec

**Review by anikiki — APPROVED** [review]
LGTM :+1:

**Pull request merged**